### ISSUE TITLE
UTC issue

### ISSUE THREAD

**Issue by pgolebiowski** [olsh/todoist-net#1]
Regarding the property `Item.DueDateUtc`, it does not seem to be in UTC when sent to Todoist. Here is your code for the converter:

```
public UtcDateTimeConverter()
{
    DateTimeFormat = "yyyy-MM-ddTHH:mm";
}
```

I think it is missing `Z` at the end: https://www.w3.org/TR/NOTE-datetime.

Even though I specifically assign a `DateTime` that has `DateTimeKind` set to `Utc` and `hour=0` and `minutes=0`, it gets converted to `01:00`, because the `UtcDateTimeConverter` does not specify UTC (missing `Z`).

**Comment by olsh** [comment]
Hi @pgolebiowski,
Thank you for reporting.

This is strange, but according to the [documentation](https://developer.todoist.com/sync/v7/#add-an-item) we shouldn't pass the `Z`. Todoist API assumes that the time is UTC.
I've just written a [test](https://github.com/olsh/todoist-net/commit/f2c675f0787786a58367bd691cadebb4b46690ec), and it seems like everything works as expected.

Please let me know if I'm wrong.

**Comment by pgolebiowski** [comment]
Hi @olsh,

First, thank you for your continuous effort on providing the library. Much appreciated! 😄 

Here is my code:

```csharp
var client = new TodoistClient("here is my token");

var item1 = new Item("Test item with date string.")
{
    DateString = "2017-12-01"
};

var item2 = new Item("Test item with UTC DateTime.")
{
    DueDateUtc = new DateTime(2017, 12, 1, 0, 0, 0, DateTimeKind.Utc)
};

var id1 = client.Items.AddAsync(item1).Result;
var id2 = client.Items.AddAsync(item2).Result;

var retrieved1 = client.Items.GetAsync(id1).Result.Item.DueDateUtc.Value;
var retrieved2 = client.Items.GetAsync(id2).Result.Item.DueDateUtc.Value;

Console.WriteLine(retrieved1.ToString("u"));
Console.WriteLine(retrieved2.ToString("u"));
```

The result:

<img width="595" alt="screen shot 2017-11-04 at 02 17 10" src="https://user-images.githubusercontent.com/7394091/32400879-6032600c-c106-11e7-88e1-5983cadaeff1.png">

```
2017-12-01 22:59:59Z
2017-12-01 00:00:00Z
```

I understand -- Todoist converts dates to the local time instead of just showing what was assigned there. It totally makes sense.

However, the result is exactly the same when I change `DateTimeKind.Utc` to `DateTimeKind.Local` 😔 I suppose your code could detect that there is `DateTimeKind.Local` set and before truncating this information, use it to obtain DateTime in UTC.

When I type:

```csharp
var item = new Item("Test item with UTC DateTime.")
{
	DueDateUtc = new DateTime(2017, 12, 1, 0, 0, 0, DateTimeKind.Local).ToUniversalTime()
};

var id = client.Items.AddAsync(item).Result;
var retrieved = client.Items.GetAsync(id).Result.Item.DueDateUtc.Value;
Console.WriteLine(retrieved.ToString("u"));
```

Then it works:

<img width="586" alt="screen shot 2017-11-04 at 02 38 08" src="https://user-images.githubusercontent.com/7394091/32401112-367f53d4-c109-11e7-82df-b8d7a27d28ad.png">

```
2017-11-30 23:00:00Z
```

Hope it helps.

**Comment by olsh** [comment]
Ahh, I see the problem now.
Completely agree with you that it should convert local time to UTC. Will fix this soon.

**Comment by olsh** [comment]
Thank you for the detail explanation! 👍 
Just published 1.3.3 with the bugfix.

**Comment by pgolebiowski** [comment]
@olsh Glad to help. Keep up the great work! 😄
